homeassistant/components/russound_rio/config_flow.py

```diff
@@ -0,0 +1,107 @@
+    raise NoPrimaryControllerException
+
+
+class FlowHandler(ConfigFlow, domain=DOMAIN):
+    """Russound RIO configuration flow."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initialized by the user."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            port = user_input[CONF_PORT]
+
+            controllers = None
+            russ = Russound(self.hass.loop, host, port)
+            try:
+                async with asyncio.timeout(5):
+                    await russ.connect()
+                    controllers = await russ.enumerate_controllers()
+                    metadata = find_primary_controller_metadata(controllers)
+                    await russ.close()
+            except RUSSOUND_RIO_EXCEPTIONS:
+                _LOGGER.exception("Could not connect to Russound RIO")
+                errors["base"] = "cannot_connect"
+            except NoPrimaryControllerException:
+                _LOGGER.exception(
+                    "Russound RIO device doesn't have a primary controller",
+                )
+                errors["base"] = "no_primary_controller"
+            else:
+                await self.async_set_unique_id(metadata[0])
+                data = {CONF_HOST: host, CONF_PORT: port, CONF_MODEL: metadata[1]}
+                return self.async_create_entry(title=metadata[1], data=data)
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_import(
+        self, import_config: dict[str, Any]
+    ) -> ConfigFlowResult:
+        """Attempt to import the existing configuration."""
+        self._async_abort_entries_match({CONF_HOST: import_config[CONF_HOST]})
+        host = import_config[CONF_HOST]
+        port = import_config.get(CONF_PORT, 9621)
+
+        # Connection logic is repeated here since this method will be removed in future releases
+        russ = Russound(self.hass.loop, host, port)
+        try:
+            async with asyncio.timeout(5):
+                await russ.connect()
+                controllers = await russ.enumerate_controllers()
+                metadata = find_primary_controller_metadata(controllers)
+                await russ.close()
+        except RUSSOUND_RIO_EXCEPTIONS:
+            _LOGGER.exception("Could not connect to Russound RIO")
+            return self.async_abort(
+                reason="cannot_connect", description_placeholders={}
+            )
+        except NoPrimaryControllerException:
+            _LOGGER.exception("Russound RIO device doesn't have a primary controller")
+            return self.async_abort(
+                reason="no_primary_controller", description_placeholders={}
+            )
+        else:
+            await self.async_set_unique_id(metadata[0])
```

Comment:
Let's also abort if we already have this unique_id (but leave the parameters empty :) )